src/lib.rs

```diff
@@ -906,13 +908,31 @@ impl Pandoc {
         // apply all filters
         let filtered = filters.into_iter().fold(o, |acc, item| item(acc));
         self.input = Some(InputKind::Pipe(filtered));
+        Ok(())
+    }
+    
+    /// actually execute pandoc
+    pub fn execute(mut self) -> Result<(), PandocError> {
+        let _ = try!(self.preprocess());
         let _ = try!(self.run());
         Ok(())
     }
+
+    /// Execute pandoc and return the result from stdout.
+    pub fn execute_with_output(mut self) -> Result<String, PandocError> {
```

Comment:
An alternative approach here (since this PR already proposes breaking changes): I could just combine `execute` and `execute_with_output`, and document (or define a type!) to indicate be what ends up in `stdout`.

For example, I could define an enum, and return that in the result, so:

```rust
/// The result of a pandoc operation
PandocOutput {
    /// The results of the pandoc operation are stored in `Path`
    ToFile(Path),
    /// The results of the pandoc operation are returned as a `String`
    ToBuffer(String),
}

pub fn execute(mut self) -> Result<PandocOutput, PandocError> {
    /* ... */
}
```